### PR TITLE
Fix some more PHPMD warnings

### PR DESCRIPTION
**Pull request by realityking**
and deprecate some stuff so they'll be fixed in the near future.

### CODE REVIEW

File: libraries/joomla/installer/installer.php [hunk 1]
```diff
@@ -143,9 +143,23 @@ public static function getInstance()
 	 * @return  boolean  Allow overwrite switch
 	 *
 	 * @since   11.1
+	 * @deprectaed 12.1 Use JInstaller::isOverwrite()
 	 */
 	public function getOverwrite()
 	{
+		JLog::add('JInstaller::getOverwrite() is deprectaed. Please use JInstaller::isOverwrite() instead', JLog::WARNING, 'deprecated');
+		return self::isOverwrite();
```
Comment (chdemko) [inline libraries/joomla/installer/installer.php]: should it be

```
return $this->isOverwrite();
```

?

File: libraries/joomla/installer/installer.php [hunk 2]
```diff
@@ -206,9 +220,23 @@ public function setRedirectURL($newurl)
 	 * @return  boolean
 	 *
 	 * @since   11.1
+	 * @deprectaed 12.1 Use JInstaller::isUpgrade()
 	 */
 	public function getUpgrade()
 	{
+		JLog::add('JInstaller::getUpgrade() is deprectaed. Please use JInstaller::isUpgrade() instead', JLog::WARNING, 'deprecated');
+		return self::isUpgrade();
```
Comment (chdemko) [inline libraries/joomla/installer/installer.php]: should it be

```
return $this->isUpgrade();
```

?

Comment (realityking) [inline libraries/joomla/installer/installer.php]: fixed

File: libraries/joomla/installer/installer.php (hunk 1 above)
Comment (realityking) [inline libraries/joomla/installer/installer.php]: fixed

**Pull request merged**